packages/grid/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts

```diff
@@ -0,0 +1,237 @@
+import * as React from 'react';
+import {
+  useGridApiEventHandler,
+  GridFeatureModeConstant,
+  GridRenderedRowsIntervalChangeParams,
+  useGridSelector,
+  GridRowId,
+  gridVisibleSortedRowIdsSelector,
+  gridSortModelSelector,
+  gridFilterModelSelector,
+  useGridApiOptionHandler,
+  GridEventListener,
+  GridRowEntry,
+  GridDimensions,
+  GridFeatureMode,
+} from '@mui/x-data-grid';
+import { useGridVisibleRows } from '@mui/x-data-grid/internals';
+import { getRenderableIndexes } from '@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller';
+import { GridApiPro } from '../../../models/gridApiPro';
+import {
+  DataGridProProcessedProps,
+  GridExperimentalProFeatures,
+} from '../../../models/dataGridProProps';
+import { GRID_SKELETON_ROW_ROOT_ID } from './useGridLazyLoaderPreProcessors';
+import { GridFetchRowsParams } from '../../../models/gridFetchRowsParams';
+
+function findSkeletonRowsSection(
+  visibleRows: GridRowEntry<any>[],
+  range: { firstRowIndex: number; lastRowIndex: number },
+): { firstRowIndex: number; lastRowIndex: number } {
+  const visibleRowsSection = visibleRows.slice(range.firstRowIndex, range.lastRowIndex);
+  let firstRowIndex = range.firstRowIndex;
+  let lastRowIndex = range.lastRowIndex;
+  let startIndex = 0;
+  let endIndex = visibleRowsSection.length - 1;
+  let isSkeletonSectionFound = false;
+
+  while (!isSkeletonSectionFound && firstRowIndex < lastRowIndex) {
+    if (!visibleRowsSection[startIndex].model && !visibleRowsSection[endIndex].model) {
```

Comment:
Side note: at some point we might need to introduce invalidation mechanizm, so that rows that were already fetched won't get fetched again, unless certain amount of time have passed.

Reply:
That is currently the case with the exception of the time variable. It is a good suggestion for improvement.